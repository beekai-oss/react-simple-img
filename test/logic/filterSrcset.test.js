import filterImgSrc from '../../src/logic/filterSrcset';

jest.mock('../../src/utils/parseSrcset', () => () => []);
jest.mock('../../src/utils/findClosestDpr', () => () => ({ url: 'test' }));

describe('filterImgSrc', () => {
  it('should produce the correct image path', () => {
    expect(
      filterImgSrc({ srcset: 'test.jpg 2x, test-small.jpg', dataset: { src: 'test.jpg', srcset: 'test.jpg' } }),
    ).toMatchSnapshot();
  });

  it('should return src if srcset not been set', () => {
    expect(filterImgSrc({ dataset: { src: 'test' } })).toBe('test');
  });
});
