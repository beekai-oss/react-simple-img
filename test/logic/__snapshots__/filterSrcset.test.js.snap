// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterImgSrc should produce the correct image path 1`] = `"test"`;
